From 0449c7f7326c96d8ac2fd82fcdb8702c6bdbdd1b Mon Sep 17 00:00:00 2001
From: Josua Mayer <josua@solid-run.com>
Date: Sun, 7 Apr 2024 16:25:43 +0200
Subject: [PATCH 8/9] HACK: arm: dts: armada-cp110: disambiguate mdio bus names
 with cp name

Board code can use the device-tree node name (e.g. mdio@12a200) to
perform ethernet phy fixups. On systems with multiple CPs, e.g. CN9131,
this name collides between all CPs.
Prepend cp-name to disambiguate (e.g. cp0-mdio).

This is a hack in that it violates device-tree common naming scheme.
Better would be introduction of a new mii_dev getter taking device-tree
path instead of name.

Signed-off-by: Josua Mayer <josua@solid-run.com>
---
 arch/arm/dts/armada-cp110.dtsi | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm/dts/armada-cp110.dtsi b/arch/arm/dts/armada-cp110.dtsi
index d3e18ec24f8..4bbe31530cf 100644
--- a/arch/arm/dts/armada-cp110.dtsi
+++ b/arch/arm/dts/armada-cp110.dtsi
@@ -34,7 +34,7 @@
 			interrupt-parent = <&gic>;
 			ranges = <0x0 U64_TO_U32_H(CP110_BASE) U64_TO_U32_L(CP110_BASE) 0x2000000>;
 
-			CP110_LABEL(mdio): mdio@12a200 {
+			CP110_LABEL(mdio): CP110_LABEL(mdio)@12a200 {
 				#address-cells = <1>;
 				#size-cells = <0>;
 				compatible = "marvell,orion-mdio";
@@ -43,7 +43,7 @@
 				status = "disabled";
 			};
 
-			CP110_LABEL(xmdio): mdio@12a600 {
+			CP110_LABEL(xmdio): CP110_LABEL(xmdio)@12a600 {
 				#address-cells = <1>;
 				#size-cells = <0>;
 				compatible = "marvell,xmdio";
-- 
2.43.0

